### PR TITLE
Update outdated default image for AWS/GCP

### PR DESCRIPTION
**Pull request by guessi**
- Fix all regions have corresponding default AMI mapping.
- Refresh outdated default images for AWS & GCP.
- Change default instance type with latest `T3` series.
- Support for AWS new region: `ap-southeast-3`, `Asia Pacific (Jakarta)`

AMI information comes from upstream Ubuntu Cloud Image Finder
- https://cloud-images.ubuntu.com/locator/

Or alternative, retrieve from scripts:

**_AWS:_**
```
curl -fsSL https://cloud-images.ubuntu.com/locator/releasesTable | \
    awk -F '"' '/Amazon.*focal.*amd64.*20220706*/{print$4" "$18}' | \
    sed 's/\>//;s/\<\/a\>//' | \
    column -t | \
    sort
```

**_GCP:_**
```
curl -fsSL https://cloud-images.ubuntu.com/locator/releasesTable | \
    awk -F '"' '/Google.*amd64.*ubuntu-2004-focal-v20220712.*/{print$4" "$16}' | \
    column -t | \
    sort
```

**_Last Update: 2022.07.16_**

### CODE REVIEW

**Comment by mtb-xt** [comment]
@bradrydzewski @tboerger is there any chance we could merge this? It would be awesome if we could have AWS SSM agent in our drone agent instances...

**Review by tboerger — APPROVED** [review]

**Comment by ademariag** [comment]
Any chance that we could push this at one point?

**Comment by guessi** [comment]
rebase, and update to current latest images

**Comment by ademariag** [comment]
@bradrydzewski @tboerger is there any chance we could merge this?

**Comment by tboerger** [comment]
I can't merge on this repo, it's up to @bradrydzewski

**Comment by ademariag** [comment]
Is this project dead? 
I don't see any updates lately and lots of valuable contributions are not being accepted. @bradrydzewski could you add more maintainers perhaps from the community?

**Comment by guessi** [comment]
Looks like there are less activity 2020-Q3. Not sure if it is related to [Drone.io Joins the Harness Family](https://www.prnewswire.com/news-releases/harness-acquires-continuous-integration-pioneer-droneio-and-commits-to-open-source-301106473.html) ?

**Comment by guessi** [comment]
FYI, just update the default images available on AWS/GCP

**Review by bradrydzewski — DISMISSED** [review]

**Comment by guessi** [comment]
Two years later, it finally got approved 🎉 

For long-term consideration, it feels like we need an auto discovery mechanism for discovering new image id for specific requirement of the platform. Those image ids are now 6 month old 👶 🍼 since my last commit.

_Let me know if image ids update is required before it get merged._

**Comment by guessi** [comment]
FYI, just refreshed AMIs, and added AWS new region: `ap-southeast-3`, `Asia Pacific (Jakarta)`.

**Comment by bradrydzewski** [comment]
@guessi thanks, can you please provide the link where you are sourcing the AMI list?  For security purposes, we need to manually verify before we can merge (sorry, this is a security requirement, trust but verify ...)

**Comment by guessi** [comment]
@bradrydzewski sure, should I put the link here, or as comment in code?

**Comment by bradrydzewski** [comment]
I noticed these images are being labeled as ssd, but the autoscaler does not use ssd disk types by default.  Does this create a possible regressions?

**Comment by guessi** [comment]
@bradrydzewski I personally only use SSD nowadays and I have used these image as source for a long time :-)

If this is really a concern, we can alternatively retrieve image id from awscli instead, which might be much reliable for user?

```
aws ssm get-parameters \
    --names /aws/service/canonical/ubuntu/server/20.04/stable/current/amd64/hvm/ebs-gp2/ami-id
```

From trustworthy source mentioned in Ubuntu Official Documentation[1].

or

```
aws ec2 describe-images \
    --owners aws-marketplace \
    --filters "Name=owner-id,Values=679593333241" "Name=name,Values=*ubuntu-focal-20.04-amd64*" \
    --query "sort_by(Images, &CreationDate)[-1:].[Name, ImageId]" \
    --region us-east-1 \
    --output text
```

Where "679593333241" is Canonical's ID, where you can find how to verify it is released by Canonical here[2].

Or even more better, retrieve from AWS API.

Ref:
- [1] https://ubuntu.com/server/docs/cloud-images/amazon-ec2
- [2] https://blog.gruntwork.io/7616fe46b49a

**Comment by bradrydzewski** [comment]
I have no problem with ssd or even ssd by default, I just want to make sure that if someone runs the autoscaler with default options it will work out of the box and not error.  If we use images labeled hvm-ssd do we need to make any changes to volume type or will everything _just work_?  I am asking because I do not use AWS so the answer to this is non-obvious to me.

**Comment by guessi** [comment]
Your concern is not exist, since `hvm-ssd` have been used for quite long time.

To verify that we can easily find existed image ids for `us-east-1`[1], `us-west-1`[2] and you will find that current images are `hvm-ssd` already.

AWS US-EAST-1
```
$ aws ec2 describe-images --region us-east-1 --image-ids "ami-43a15f3e" --output json | grep "ImageLocation" # image id for us-east-1
"ImageLocation": "099720109477/ubuntu/images/hvm-ssd/ubuntu-xenial-16.04-amd64-server-20180306",
```

AWS US-WEST-1
```
% aws ec2 describe-images --region us-west-1 --image-ids "ami-925144f2" --output json | grep "ImageLocation" # image id for us-west-1
"ImageLocation": "099720109477/ubuntu/images/hvm-ssd/ubuntu-xenial-16.04-amd64-server-20180306",
```

- [1] Image ID for us-east-1: https://github.com/drone/autoscaler/blob/810c77f/drivers/amazon/util.go#L42
- [2] Image ID for us-west-1: https://github.com/drone/autoscaler/blob/810c77f/drivers/amazon/util.go#L47

** `810c77f` is current master commit hash.

**Comment by bradrydzewski** [comment]
> you will find that current images are hvm-ssd already.

thanks!  I think this is the context I was missing

**Comment by guessi** [comment]
Just update once again after month.

**Review by bradrydzewski — APPROVED** [review]

**Comment by guessi** [comment]
Umm... looks like I need to resolve conflict and bump the image version again.

File: drivers/amazon/util.go
Comment (guessi) [inline drivers/amazon/util.go:78]: Current latest version available from upstream Ubuntu. Yes, it's "2021", but it's latest.

**Comment by guessi** [comment]
Just update the the image id for **_AWS_** and **_GCP_** once again.

At the moment of  I created this PR (Sep 11, 2020)  the latest version available was **_Ubuntu Server 20.04 LTS_** and 2 years later, it's now 2022, Ubuntu Server 22.04 LTS was [released on Apr 22, 2022](https://ubuntu.com/blog/ubuntu-22-04-lts-whats-new-linux-desktop).

Feel free to let me know if it is needed to update to **_Ubuntu Server 22.04 LTS_**.

**Comment by maxgruebneraeroqual** [comment]
We've run into an issue with builds that was sorted by switching to Ubuntu 22.04 AMIs.

https://community.harness.io/t/drone-autoscaler-amis-too-old-for-net-6-application/12249/5

Further, we've just got an email from AWS about the 16.04 Ubuntu AMIs being deprecated.

**Comment by tphoney** [comment]
thanks for the continued work on this PR !

solid work !!

**Pull request merged**
